src/Cli/dotnet/commands/dotnet-test/TestApplication.cs

```diff
@@ -18,6 +18,7 @@ internal sealed class TestApplication : IDisposable
 
         private NamedPipeServer _pipeConnection;
         private Task _namedPipeConnectionLoop;
+        private HashSet<string> _executionIds = [];
```

Comment:
This one could be accessed in parallel right?

Reply:
> The code looks pointless anyway: the only use of `_executionIds` is the `_executionIds.Add(executionId)` call; nothing enumerates the HashSet or calls `Contains`. Is a separate pull request planned that would start reading the execution IDs from `_executionIds`?

Yes, exactly. There will be another PR that makes use of this Hashset.